strawberryfields/api/connection.py

```diff
@@ -227,7 +227,7 @@ def get_job_result(self, job_id: str) -> Result:
             with io.BytesIO() as buf:
                 buf.write(response.content)
                 buf.seek(0)
-                samples = np.load(buf)
+                samples = np.load(buf, allow_pickle=False)
```

Comment:
This is great! What sort of error did this raise before?

Reply:
It seems like `allow_pickle` is `False` by default. Have you added this as a precaution/to be explicit regarding not accepting anything other than not loading object arrays, or is there another reason for this?

Reply:
Spot on @thisac. The issue is more that `allow_pickle=False` was only made the default in NumPy v1.17; in <=1.16, it's True by default!

While SF pins `numpy==1.17.4`, `pip` is very bad at dependency resolving, and it's possible that a subsequent installation could downgrade numpy --- this just plugs this as a potential security hole.

Reply:
That makes perfect sense. Thanks for the explanation (and a really nice security fix).